packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx

```diff
@@ -65,6 +93,68 @@ export const PaymentUpdateForm = ({
     resetUpdatePaymentIsSuccess,
   ]);
 
+  const {
+    exp_month,
+    exp_year,
+    payment_provider,
+    paypal_payment_error,
+  } = customer;
+
+  const actionButton = (
+    <ActionButton
+      {...{ customer, onRevealUpdateClick, revealFixPaymentModal }}
+    />
+  );
+
+  const errorAlertBarContent = (
+    <Localized
+      id="sub-route-invalid-payment-alert"
+      elems={{ div: actionButton }}
+    >
+      <span>
+        Invalid payment information, there is an error with your account.{' '}
+        {actionButton}
+      </span>
+    </Localized>
+  );
+
+  const { config } = useContext(AppContext);
+  const [paypalScriptLoaded, setPaypalScriptLoaded] = useState(false);
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (!config.featureFlags.usePaypalUIByDefault) {
+      return;
+    }
+
+    if (paypalButtonBase) {
+      setPaypalScriptLoaded(true);
+      return;
+    }
+
+    // Read nonce from the fxa-paypal-csp-nonce meta tag
+    const cspNonceMetaTag = document?.querySelector(
+      'meta[name="fxa-paypal-csp-nonce"]'
+    );
+    const cspNonce = JSON.parse(
+      decodeURIComponent(cspNonceMetaTag?.getAttribute('content') || '""')
+    );
+
+    const script = document.createElement('script');
+    script.src = `${config.paypal.scriptUrl}/sdk/js?client-id=${config.paypal.clientId}&vault=true&commit=false&intent=capture&disable-funding=credit,card`;
+    // Pass the csp nonce to paypal
+    script.setAttribute('data-csp-nonce', cspNonce);
+    /* istanbul ignore next */
+    script.onload = () => {
+      setPaypalScriptLoaded(true);
+    };
+    /* istanbul ignore next */
+    script.onerror = () => {
+      throw new Error('Paypal SDK could not be loaded.');
+    };
+    document.body.appendChild(script);
+  }, []);
```

Comment:
We should dedupe this hook; maybe export it from the `PaypalButton` component, so it can be used in `SubscriptionCreate` as well as `PaymentUpdateForm`?